src/ImageSharp/IO/BufferedReadStream.cs

```diff
@@ -213,20 +213,13 @@ public override void Flush()
     [MethodImpl(MethodImplOptions.AggressiveInlining)]
     public override long Seek(long offset, SeekOrigin origin)
     {
-        switch (origin)
+        this.Position = origin switch
```

Comment:
Hmmm.... I think we should have done the opposite and removed any casting from the chunked stream. There's potential for overflow here.